tracing-appender/src/rolling.rs

```diff
@@ -500,13 +508,75 @@ impl Inner {
                     .unwrap_or(0),
             ),
             rotation,
+            keep_last: None,
         };
         (inner, writer)
     }
 
+    fn prune_old_logs(&self, keep_last: usize) {
+        let files = fs::read_dir(&self.log_directory).map(|dir| {
+            dir.filter_map(Result::ok)
+                .filter(|entry| {
+                    entry
+                        .file_name()
+                        .to_string_lossy()
+                        .contains(&self.log_filename_prefix)
+                })
+                .collect::<Vec<_>>()
+        });
+
+        let format;
+        match self.rotation {
+            Rotation::MINUTELY => {
+                format = format_description::parse("[year]-[month]-[day]-[hour]-[minute]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::HOURLY => {
+                format = format_description::parse("[year]-[month]-[day]-[hour]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::DAILY => {
+                format = format_description::parse("[year]-[month]-[day]")
+                    .expect("Unable to create a formatter; this is a bug in tracing-appender");
+            }
+            Rotation::NEVER => {
+                unreachable!("keep last N files feature is not available for `Rotation::NEVER`!")
+            }
+        }
+
+        match files {
+            Ok(files) => {
+                if files.len() > (keep_last - 1) {
+                    let mut sorted_files = vec![];
+                    for file in files {
+                        let filename = file.file_name().to_string_lossy().to_string();
+                        let date_str = filename[(filename.rfind('.').unwrap() + 1)..].to_string();
```

Comment:
Are you sure this will never panic?

Reply:
As long as the format of adding dates will stay the same, this should not panic imo. 
Currently, the date is added to the filename with a `.`, so there should be a `.` just before the date :)

Reply:
I see! Probably an edge case, but imagine :
- my prefix doesn’t contain any extension (say for instance: "abc"). `tracing-appender` will generate files such as `abc.2022.06.03` and so on and so forth.
- however, user can create a file called `abc`
- this file will be collected by the code at line 517, because the predicate is true ("abc" is contained in "abc")
- `rfind(’.’)` will returns `None`

Quite an edge case though, maybe not worth addressing?

Reply:
I see. Yes, if the user manipulates the log folder, everything becomes possible :/ 
Let me try to handle it with a better error handling! Thank you very much for the suggestion